pkg/reconciler/kubernetes/tektonresult/transform.go

```diff
@@ -0,0 +1,209 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tektonresult
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"knative.dev/pkg/logging"
+
+	mf "github.com/manifestival/manifestival"
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	"github.com/tektoncd/operator/pkg/reconciler/common"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	// Results ConfigMap
+	ConfigAPI        = "tekton-results-api-config"
+	DeploymentAPI    = "tekton-results-api"
+	ConfigINFO       = "tekton-results-info"
+	ConfigMetrics    = "tekton-results-config-observability"
+	ConfigPostgresDB = "tekton-results-postgres"
+	pvcLoggingVolume = "tekton-logs"
+)
+
+// transform mutates the passed manifest to one with common, component
+// and platform transformations applied
+func (r *Reconciler) transform(ctx context.Context, manifest *mf.Manifest, comp v1alpha1.TektonComponent) error {
+	instance := comp.(*v1alpha1.TektonResult)
+	targetNs := comp.GetSpec().GetTargetNamespace()
+	extra := []mf.Transformer{
+		common.InjectOperandNameLabelOverwriteExisting(v1alpha1.OperandTektoncdPipeline),
+		common.ApplyProxySettings,
+		common.ReplaceNamespaceInDeploymentArgs(targetNs),
+		common.ReplaceNamespaceInDeploymentEnv(targetNs),
+		updateApiConfig(instance.Spec.ResultsAPIProperties),
+		enablePVCLogging(ctx, instance.Spec.ResultsAPIProperties),
+		common.AddDeploymentRestrictedPSA(),
+		common.AddStatefulSetRestrictedPSA(),
+	}
+	extra = append(extra, r.extension.Transformers(instance)...)
+	return common.Transform(ctx, manifest, instance, extra...)
+}
+
+// ReplaceNamespaceInDeploymentArgs replaces namespace in deployment's args
+func enablePVCLogging(ctx context.Context, p v1alpha1.ResultsAPIProperties) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "Deployment" || u.GetName() != DeploymentAPI || p.LOGGING_PVC_NAME == "" {
+			return nil
+		}
+
+		logger := logging.FromContext(ctx)
+		d := &appsv1.Deployment{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, d)
+		if err != nil {
+			return err
+		}
+
+		set := true
+		for i := 0; i < len(d.Spec.Template.Spec.Containers[0].VolumeMounts); i++ {
+			if d.Spec.Template.Spec.Containers[0].VolumeMounts[i].Name == pvcLoggingVolume {
+				d.Spec.Template.Spec.Containers[0].VolumeMounts[i] = corev1.VolumeMount{Name: pvcLoggingVolume,
+					MountPath: p.LOGS_PATH}
+				logger.Debugf("Volumemounts set for tekton results logging")
+				set = false
+			}
+		}
+		if set {
+			d.Spec.Template.Spec.Containers[0].VolumeMounts = append(d.Spec.Template.Spec.Containers[0].VolumeMounts,
+				corev1.VolumeMount{Name: pvcLoggingVolume,
+					MountPath: p.LOGS_PATH})
+		}
+
+		set = true
+		vol := corev1.Volume{
```

Comment:
move `vol` variable creation to inside `if set {` block, line number 111

Reply:
We are using this both in for as well as if block.